### PR TITLE
Add support for management of User crontabs

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
@itgrl: The only detail that is missing are the unit tests for your new define.
Please let me know if you need help or have questions :)

**Comment by itgrl** [comment]
@Phil-Friderici  It appears that there are failures attempting to clone the repo for the checks.  This doesn't seem to be issues with the rspec tests that I have added.  Should I push another commit to trigger auto-testing again?

**Comment by Phil-Friderici** [comment]
The connection between GitHub and Travis seems to be unstable today. Have seen the same behaviour in other cases today. Have retriggered the failed jobs and they all got green :)

File: spec/defines/crontab_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    
+    context 'with vars set' do
+      let (:params) do
+        {
+          :ensure  => 'file',
+          :owner   => 'operator',
+          :group   => 'operator',
+          :mode    => '0242',
+          :content => 'SHELL=/bin/bash',
+        }
```
Comment (Phil-Friderici) [inline spec/defines/crontab_spec.rb]: actually `entries` or `vars` aren't set in these test cases

File: manifests/user/crontab.pp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#          vars:
+#            'SHELL': '/bin/bash'
+#            'MYECHO': '$(which echo)'
+#          entries:
+#            '# Echo Hello World': '* 3 * * * $MYECHO "Hello user2!" 2>&1'
+#
+#
+#
+define cron::user::crontab (
+  $ensure  = file,
```
Comment (Phil-Friderici) [inline manifests/user/crontab.pp]: `ensure` doesn't bring functionallity as it isn't used in the file resource further down.

File: manifests/user/crontab.pp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  if is_string($owner) == false { fail('cron::user::crontab::owner must be a string') }
+  if is_string($group) == false { fail('cron::user::crontab::group must be a string') }
+
+  validate_absolute_path($path_real)
+  validate_re($ensure, '^(absent|file|present)$', "cron::fragment::ensure is ${ensure} and must be absent, file or present")
+  validate_re($mode, '^[0-7]{4}$', "cron::fragment::mode is <${mode}> and must be a valid four digit mode in octal notation.")
+
+  $content_real = $content ? {
+    undef   => "template('cron/crontab.erb')",
```
Comment (Phil-Friderici) [inline manifests/user/crontab.pp]: Doubleticks will prevent using the template. Should read:
```
undef   => template('cron/crontab.erb'),
```

Fixed in one of the commits I'll send to you soon.

File: spec/defines/crontab_spec.rb (hunk 1 above)
Comment (Phil-Friderici) [reply spec/defines/crontab_spec.rb]: fixed in https://github.com/itgrl/puppet-module-cron/pull/3

File: manifests/user/crontab.pp (hunk 3 above)
Comment (Phil-Friderici) [reply manifests/user/crontab.pp]: fixed in https://github.com/itgrl/puppet-module-cron/pull/3

**Comment by Phil-Friderici** [comment]
By default the default for `MAILTO` is set to root. I am not a cron pro, but I guess that should be equal to the username. Guess it would be a good idea to parameterize that variable in the template.

**Comment by Phil-Friderici** [comment]
Travis should run successfully and I think we are ready :)
Once Travis finished successfull, please squash all the commits into one (or a smaller amount) as last step before I can merge the new functionality.

**Comment by Phil-Friderici** [comment]
rebased all these commits into two commits in https://github.com/Ericsson/puppet-module-cron/pull/55

**Comment by Phil-Friderici** [comment]
@itgrl Released as 2.3.0. Thanks for your work!

**Pull request closed without merging**